src/components/FormElements/XdrLedgerKeyPicker.tsx

```diff
@@ -0,0 +1,406 @@
+      }, {} as AnyObject);
+
+      if (selectedLedgerKey.addlFields) {
+        formLedgerKeyJson[selectedLedgerKey.id] = selectedLedgerKey.addlFields;
+      }
+
+      const formLedgerKeyJsonString = JSON.stringify(formLedgerKeyJson);
+      setLedgerKeyJsonString(formLedgerKeyJsonString);
+    }
+  }, [ledgerKeyJsonString, selectedLedgerKey]);
+
+  const renderLedgerKeyTemplate = () => {
+    if (!selectedLedgerKey || !ledgerKeyJsonString) {
+      return null;
+    }
+
+    return selectedLedgerKey.templates.split(",").map((template) => {
+      const ledgerKeyJson = JSON.parse(ledgerKeyJsonString);
+      const json = JSON.parse(ledgerKeyJsonString);
+      const obj = json[selectedLedgerKey!.id];
+
+      const component = formComponentTemplateEndpoints(template);
+
+      if (component) {
+        const handleChange = (key: any, val: any) => {
+          const error = component.validate?.(val);
+
+          if (error) {
+            setFormError(error);
+          } else {
+            setFormError("");
+          }
+
+          obj[key] = val;
+
+          const string = JSON.stringify(json);
+          setLedgerKeyJsonString(string);
+
+          const jsonXdrEncoded = jsonEncodeXdr(string);
+
+          if (jsonXdrEncoded?.xdrString) {
+            setLedgerKeyXdr(jsonXdrEncoded.xdrString);
+          }
+        };
+
+        if (template === "asset") {
+          return component.render({
+            value: ledgerKeyJson[selectedLedgerKey.id][template],
+            error: formError,
+            onChange: (assetObjVal: AssetObjectValue) => {
+              handleChange(
+                assetObjVal,
+                isEmptyObject(sanitizeObject(assetObjVal || {}))
+                  ? undefined
+                  : JSON.stringify(assetObjVal),
+              );
+            },
+            isRequired: true,
+          });
+        }
+
+        return component.render({
+          value: ledgerKeyJson[selectedLedgerKey.id][template],
+          error: formError,
+          onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+            handleChange(template, e.target.value);
+          },
+          isRequired: true,
+        });
+      }
+      return null;
+    });
+  };
+
+  return (
+    <Box gap="sm" key={id}>
+      <XdrPicker
+        id="ledger-key-xdr"
+        label="Ledger Key XDR"
+        value={value}
+        error={ledgerKeyXdrError}
+        onChange={(e) => {
+          onChange(e.target.value);
+        }}
+      />
+      <Select
+        id={`${id}-type`}
+        fieldSize="md"
+        label="Ledger Key"
+        value={selectedLedgerKey?.id}
+        onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+          const selectedVal = e.target.value;
+          setLedgerKeyJsonString("");
+
+          if (selectedVal) {
+            const selectedLedgerKey = getKeyType(selectedVal);
+            selectLedgerKey(selectedLedgerKey!);
+          } else {
+            selectLedgerKey(null);
+          }
```

Comment:
Just a suggestion in passing: I'm a sucker for one-liners lol but feel free to ignore

```suggestion
          selectLedgerKey(selectedVal ? getKeyType(selectedVal)! : null);
```

Reply:
ty!